package.json

```diff
@@ -144,12 +144,15 @@
     "vadimcn.vscode-lldb"
   ],
   "scripts": {
-    "vscode:prepublish": "npm run compile",
-    "compile": "tsc -p ./",
-    "watch": "tsc -watch -p ./",
-    "pretest": "npm run compile && npm run lint",
-    "lint": "eslint src --ext ts",
+    "vscode:prepublish": "npm run esbuild-base -- --minify",
+    "esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=dist/extension.js --external:vscode --format=cjs --platform=node --target=node14",
+    "esbuild": "npm run esbuild-base -- --sourcemap",
+    "esbuild-watch": "npm run esbuild-base -- --sourcemap --watch",
+    "compile": "tsc",
+    "watch": "tsc --watch",
+    "lint": "eslint src --ext ts && tsc --noEmit",
```

Comment:
What does the `tsc --noEmit` do here?

Reply:
`esbuild` doesn't do type checking, so the recommendation was to run `tsc --noEmit` (meaning: don't emit compiler output, just type check) at the linting stage.

During development, VS Code should still flag any errors.